sos/plugins/networking.py

```diff
@@ -251,8 +227,15 @@ def setup(self):
             # Devices that exist in a namespace use less ethtool
             # parameters. Run this per namespace.
             for namespace in self.get_ip_netns(ip_netns_file):
-                for eth in self.get_netns_devs(namespace):
-                    ns_cmd_prefix = cmd_prefix + namespace + " "
+                ns_cmd_prefix = cmd_prefix + namespace + " "
+                netns_netdev_list = self.call_ext_prog(ns_cmd_prefix +
+                                                       "ls -1 /sys/class/net/")
+                for eth in netns_netdev_list['output'].splitlines():
```

Comment:
Is there a reason not to use `os.listdir()` here? It is typically cleaner than shelling out to `ls`...